.github/workflows/benchmark.yml

```
@@ -2,23 +2,58 @@ name: Run benchmarks
 
 on:
   pull_request:
+    types: [opened, synchronize, reopened, labeled]
+
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
 
 jobs:
-  Benchmark:
+  Benchmarking:
     runs-on: ubuntu-latest
     if: contains(github.event.pull_request.labels.*.name, 'performance critical')
-    env:
-      JULIA_DEBUG: BenchmarkCI
     steps:
+      # setup
       - uses: actions/checkout@v2
       - uses: julia-actions/setup-julia@latest
         with:
-          version: 1.6
-      - name: Install dependencies
-        run: julia -e 'using Pkg; pkg"add PkgBenchmark BenchmarkCI"'
-      - name: Run benchmarks
-        run: julia -e "using BenchmarkCI; BenchmarkCI.judge()"
-      - name: Post results
-        run: julia -e "using BenchmarkCI; BenchmarkCI.postjudge()"
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          version: 'nightly'
+      - uses: julia-actions/julia-buildpkg@latest
+      - name: install dependencies
+        run: julia -e 'using Pkg; pkg"add PkgBenchmark BenchmarkCI@0.1"'
+
+      # run the benchmark suite
+      - name: run benchmarks
+        run: |
+          julia -e '
+            using BenchmarkCI
+            BenchmarkCI.judge()
+            BenchmarkCI.displayjudgement()
+            '
+      # generate and record the benchmark result as markdown
+      - name: generate benchmark result
+        run: |
+          body=$(julia -e '
+          using BenchmarkCI
+          let
+              judgement = BenchmarkCI._loadjudge(BenchmarkCI.DEFAULT_WORKSPACE)
+              title = "Kernel Benchmark Result"
+              ciresult = BenchmarkCI.CIResult(; judgement, title)
+              BenchmarkCI.printcommentmd(stdout::IO, ciresult)
+          end
+          ')
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          echo $body > ./benchmark-result.artifact
```

Maybe simpler if you use a Julia script?
```suggestion
          using BenchmarkCI
          judgement = BenchmarkCI._loadjudge(BenchmarkCI.DEFAULT_WORKSPACE)
          title = "Kernel Benchmark Result"
          ciresult = BenchmarkCI.CIResult(; judgement, title)
          comment = sprint() do io
              return BenchmarkCI.printcommentmd(io, ciresult)
          end
          comment = replace(comment, "%" => "%25", "\\n" => "%0A", "\\r" => "%0D")
          write("benchmark-result.artifact", comment)
        shell: julia --color=yes {0}
```
Instead of the manual replacements maybe you could also use `Markdown.htmlesc`.